primitives/src/json.rs

```diff
@@ -0,0 +1,220 @@
+    /// - or if T contains a map with non-string keys.
+    pub fn new<T: Serialize>(payload: T) -> Self {
+        candidate::JsonCandidate::new(payload).try_into().unwrap()
+    }
+
+    /// Tries cast [`JsonString`] to any value.
+    ///
+    /// # Errors
+    /// - if invalid representation of `T`
+    pub fn try_into_any<T: DeserializeOwned>(&self) -> Result<T, serde_json::Error> {
+        serde_json::from_str(&self.0)
+    }
+
+    /// Create without checking whether the input is a valid JSON string.
+    ///
+    /// The caller must guarantee that the value is valid.
+    pub fn from_string_unchecked(value: String) -> Self {
+        Self(value)
+    }
+
+    /// Getter for [`JsonString`]
+    pub fn get(&self) -> &String {
+        &self.0
+    }
+}
+
+impl<'de> serde::de::Deserialize<'de> for JsonString {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        let json = Value::deserialize(deserializer)?;
+        Ok(Self(json.to_string()))
+    }
+}
+
+impl serde::ser::Serialize for JsonString {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        let json: Value = serde_json::from_str(&self.0).map_err(serde::ser::Error::custom)?;
+        json.serialize(serializer)
+    }
+}
+
+impl PartialOrd for JsonString {
+    fn partial_cmp(&self, other: &Self) -> Option<core::cmp::Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl PartialEq for JsonString {
+    fn eq(&self, other: &Self) -> bool {
+        self.cmp(&other).is_eq()
+    }
+}
+
+/// ToDo: Create unit tests!
+impl Ord for JsonString {
+    fn cmp(&self, other: &Self) -> core::cmp::Ordering {
+        let l: Value = serde_json::from_str(&self.0).unwrap();
+        let r: Value = serde_json::from_str(&other.0).unwrap();
+
+        match (l, r) {
+            (Value::Number(left), Value::Number(right)) => {
+                if let (Some(left), Some(right)) = (left.as_f64(), right.as_f64()) {
+                    left.total_cmp(&right)
+                } else if let (Some(left), Some(right)) = (left.as_u64(), right.as_u64()) {
+                    left.cmp(&right)
+                } else if let (Some(left), Some(right)) = (left.as_i64(), right.as_i64()) {
+                    left.cmp(&right)
+                } else {
+                    self.0.cmp(&other.0)
+                }
+            }
+            (Value::Array(left), Value::Array(right)) => {
+                let left = left.iter().map(JsonString::from);
+                let right = right.iter().map(JsonString::from);
+                left.cmp(right)
+            }
+            (Value::Object(left), Value::Object(right)) => {
+                let left = left.iter().map(|(k, v)| (k, JsonString::from(v)));
+                let right = right.iter().map(|(k, v)| (k, JsonString::from(v)));
+                left.cmp(right)
```

Comment:
I think with `serde_json` feature `preserve_order` it might give us weird results.
Like two objects like `{"a": 0, "b": 0}` and `{"b": 0, "a": 0}` won't be equal, but afaik we don't use this feature.

Reply:
after discussing with @Erigara we agreed it would be best to just derive both `Ord` and `Eq`. This custom implementation is not necessary

Reply:
In this case this test will fail `correct_pagination_assets_after_creating_new_one()`

Reply:
ouch, but I don't think this is how it should be resolved

Reply:
I think to test the solution here but before keep as is:
https://github.com/hyperledger/iroha/issues/4748

Reply:
i think we can adjust the test

Reply:
> I think to test the solution here but before keep it as is: #4748

you should derive `Eq` and `Ord`. you can ignore the test and fix in another ticket